pkg/sync/rpc/list.go

```diff
@@ -42,13 +52,24 @@ func (b *BackupList) BackupAdd(backupID string, backup *replica.BackupStatus) er
 		return fmt.Errorf("empty backupID")
 	}
 
+	// If a record already exists, remove it before adding the new one.
+	// Ignore errors - we don't really expect to find it, normally.
+	_ = b.BackupDelete(backupID)
+
 	b.Lock()
 	b.infos = append(b.infos, &BackupInfo{
 		backupID:     backupID,
 		backupStatus: backup,
 	})
 	b.Unlock()
 
+	if backupListHighWaterMark < len(b.infos) {
+		backupListHighWaterMark = len(b.infos)
```

Comment:
NIT: Should `len(b.infos)` be protected by the lock?

Reply:
I was thinking not, since the HWM is just a rough indicator of whether there is a runaway memory allocation.  We don't use the exact value for any critical decisions.  If the length changes beneath us, it will be updated the next time around.

Reply:
Sounds good.